app/src/lib/truncate-with-ellipsis.ts

```diff
@@ -0,0 +1,32 @@
+/** Truncate a single line unicode string by a given maxLength and add ellipsis if necessary */
+export function truncateWithEllipsis(str: string, maxLength: number) {
```

Comment:
This seems a great chance to write some tests for this, especially now that we have to be aware of Unicode characters in the string. Are you able to add some tests to your PR using the guide I started in https://github.com/desktop/desktop/pull/5983? I'd love some feedback on if that helps, or what else we can add.

Reply:
I wrote tests, see `app/test/unit/truncate-with-ellipsis-test.ts`. I used `chai` though following other tests as example. I'll migrate my tests to `jest` today.